src/devices/imuBosch_BNO055/imuBosch_BNO055.cpp

```diff
@@ -436,9 +435,9 @@ void BoschIMU::run()
 
     // TODO: how to optimally protect only code filling the data vector?
 
-    mutex.wait();
-
-    data.zero();
+    // In order to avoid zeros when a single read from a sensor is missing,
+    // initialize the new measure to be equal to the previous one
+    yarp::sig::Vector data_tmp = data;
```

Comment:
I agree, it is better to allocate buffers once and for all as a class member.
It'll not impact on performaces drammatically, but devices usually run on pc104 or low power computer, so any improvement easy to implement is welcome